File: WebAPI/src/myVegAppDbAPI/Model/DbModels/Place.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace myVegAppDbAPI.Model.DbModels
{
    
    [BsonIgnoreExtraElements]
    public class Place
    {
        [BsonId]
        public ObjectId Id { get; set; }

        [BsonElement("name")]
        public String Name { get; set; }

        [BsonElement("type")]
        public String Type { get; set; }

        [BsonElement("phoneNumber")]
        public String PhoneNumber { get; set; }

        [BsonElement("description")]
        public String Description { get; set; }

        [BsonElement("website")]
        public String Website { get; set; }

        [BsonElement("address")]
        public String Address { get; set; }

        [BsonElement("email")]
        public String Email { get; set; }

        [BsonElement("nReviews")]
        public Int32 nReviews { get; set; }

        [BsonElement("rating")]
        public Double? rating { get; set; }

        [BsonElement("distance")]
        public Double Distance { get; set; }
        
        [BsonElement("location")]
        public GeoLoc Location { get; set; }

        [BsonElement("menu")]
        public MenuItem[] Menu { get; set; }

        [BsonElement("country")]
        public Country Country { get; set; }

        [BsonElement("openingHours")]
        public String OpeningHours { get; set; }
    }
}

File: myApp/src/entities/place.ts

export class Place{
  _id: string;
  name:string;
  type:string;
  rating:number;
  reviews:number;
  description:string;
  position:{
    latitude:number,
    longitude:number
  };
  distance:number;
  address:string;
  phoneNumber:string;
  openingHours:string[];
  email:string;
  gallery:string[];
  website:string,
  tags:string[];

  static build(_id:string, name:string,description:string,address:string,phoneNumber:string,email:string, type:string, reviews:number,rating:number,openingHours:string[], latitude:number, longitude:number,distance:number,gallery:string[],tags:string[],website:string):Place{
    var pb = new Place();
    pb._id =_id;
    pb.name = name;
    pb.type = type;
    pb.reviews = reviews;
    pb.rating = rating;
    pb.description = description;
    pb.position = {
      latitude : latitude,
      longitude : longitude
    };
    pb.distance = distance;
    pb.gallery = gallery.map(x=>(x as any).$oid as string);
    pb.address = address
    pb.phoneNumber = phoneNumber;
    pb.email = email;
    pb.openingHours = openingHours;
    pb.tags = tags;
    return pb;
  }

  static calculateLabel(distance:number):string{
    return distance.toString();
  }
}